### PR TITLE
[TTP] Handle VM death when TTP activity in foreground

### LINKED ISSUE #8140: [Mobile Payments] Handle built-in reader disconnections and reconnection
Check if the built-in reader for Tap on Mobile will disconnect when the app goes in to the background, e.g. to take a phone call.

We should reconnect the reader when possible if this happens.

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #8140

### Description

This PR brings another error that's shown when Activity and VM are killed during TTP activity being in the foreground. Also, it adds tracking of this situation so we can determine if this is not an ideal solution and needs to be improved

### Testing instructions

Test TTP, IPP (including refund) with "don't keep activity" dev option ON and OFF

When it's on, TTP will fail with the screen that explains that this happened

### Images/gif

https://user-images.githubusercontent.com/4923871/236451752-65f053d6-8d57-491b-97c8-1f228cec2714.mp4

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt [hunk 1]
```diff
@@ -4068,6 +4068,27 @@ class CardReaderPaymentViewModelTest : BaseUnitTest() {
             verify(cardReaderTrackingInfoKeeper, never()).setCardReaderBatteryLevel(anyFloat())
         }
 
+    @Test
+    fun `given ttp in progress and reader connected, when vm starts, then AppKilledWhileInBackground state emitted`() =
```
Comment (AnirudhBhat) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt:4072]: ❓ Should we also write an invert of this test where we test if the TTP is not in progress when the VM starts, then AppKilledWhileInBackground is not emitted?

Comment (AnirudhBhat) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt:4072]: ❓ What do you think about writing a test to verify that the payment flow will not be started when the AppKilledWhileInBackground is emitted?

**Review by AnirudhBhat — APPROVED** [review]
LGTM! I've left couple of non-blocking comment.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/payments/cardreader/CardReaderPaymentViewModelTest.kt:4072]: Added both tests here 450ba439decfb730b6ad93285ec7289bd4d1538d

**Pull request merged**